MericCelik/assignment1/Q2.java

```diff
@@ -0,0 +1,152 @@
+import java.util.Scanner;
+import java.util.Stack;
+import java.util.LinkedList;
+
+public class Q2 {
+
+    /*The first solution for the problem. It uses stack to reverse the list. A recursive algorithm can
+    * be written as well.
+    *
+    * The function takes two parameter:
+    * The first one is linked list(from java.util) we will work on,
+    * The second one is the k value describing index from the last element.
+    */
+    public static Object returnLastKthElement(LinkedList<Object> givenlist, int k){
+        Stack ourstack = new Stack();
+        int sizeoflist = givenlist.size();
+        Scanner reader = new Scanner(System.in);
+
+        if(k>=sizeoflist){
+            System.out.println("K value is bigger than size of the list!");
+            return null;
+        }
+
+        else if(k<0){
+            System.out.println("K value is negative!");
+            return null;
+        }
+
+        //In the block below, the function asks for new input from user when k is inappropriate.
+        //It prevents function from returning null.
+
+        /*while(k>=sizeoflist) {
+            System.out.println("K value is bigger than size of the list! " +
+                    "Please enter a smaller value: ");
+            String ob = reader.next();
+            try{
+                k = Integer.parseInt(ob);
+            }
+            catch (NumberFormatException e)
+            {
+                System.out.println("Please enter an positive integer value!");
+                continue;
+            }
+        }
+
+        while(k<0){
+            System.out.println("K value is smaller than zero! Please enter a positive integer value!");
+            String ob = reader.next();
+            try{
+                k = Integer.parseInt(ob);
+            }
+            catch (NumberFormatException e)
+            {
+                System.out.println("Please enter an positive integer value!");
+                continue;
+            }
+        }*/
+
+        for(int i=0;i<sizeoflist;i++){
+            ourstack.push(givenlist.get(i));
```

Comment:
optional: While the complexity (O(n)) is very good, thi solution uses twice as much memory than it needs (copies everything from the list to the stack). Can you think of a solution (also in O(n)) that would only use max 2 LinkedListNodes?

Reply:
I added two more solutions working on O(n) in worst (one iterative and one recursive). I will try to find the algorithm you mentioned!